src/wizards/clusters.ts

```diff
@@ -0,0 +1,133 @@
+        state.bootstrap = await input.showInputBox({
+            title: INPUT_TITLE,
+            step: input.getStepNumber(),
+            totalSteps: state.totalSteps,
+            value: state.bootstrap ? state.bootstrap : DEFAULT_BROKER,
+            prompt: 'Broker(s) (localhost:9092,localhost:9093...)',
+            validate: validateBroker
+        });
+        return (input: MultiStepInput) => inputClusterName(input, state, clusterSettings);
+    }
+
+    async function inputClusterName(input: MultiStepInput, state: Partial<AddClusterState>, clusterSettings: ClusterSettings) {
+        const existingClusterNames = clusterSettings.getAll().map(cluster => cluster.name);
+        state.name = await input.showInputBox({
+            title: INPUT_TITLE,
+            step: input.getStepNumber(),
+            totalSteps: state.totalSteps,
+            value: state.name ||  '',
+            prompt: 'Friendly name',
+            validationContext: existingClusterNames,
+            validate: validateClusterName
+        });
+
+        return (input: MultiStepInput) => inputAuthentification(input, state);
+    }
+
+    async function inputAuthentification(input: MultiStepInput, state: Partial<AddClusterState>) {
+        const authOptions: QuickPickItem[] = [{ "label": "None" }, { "label": "SASL/PLAIN" }];
+        const authentification = (await input.showQuickPick({
+            title: INPUT_TITLE,
+            step: input.getStepNumber(),
+            totalSteps: state.totalSteps,
+            placeholder: 'Pick authentification',
+            items: authOptions,
+            activeItem: authOptions[0]
+        })).label;
+        if (authentification && authentification == authOptions[1].label) {
+            state.saslOption = { mechanism: 'plain' };
+            return (input: MultiStepInput) => inputAuthentificationUserName(input, state);
+        }
+        return undefined;
+    }
+
+    async function inputAuthentificationUserName(input: MultiStepInput, state: Partial<AddClusterState>) {
+        if (!state.saslOption) {
+            return;
+        }
+        state.saslOption.username = await input.showInputBox({
+            title: INPUT_TITLE,
+            step: input.getStepNumber(),
+            totalSteps: state.totalSteps,
+            value: state.saslOption?.username || '',
+            prompt: ' Username',
+            validate: validateAuthentificationUserName
+        });
+
+        return (input: MultiStepInput) => inputAuthentificationPassword(input, state);
+    }
+
+    async function inputAuthentificationPassword(input: MultiStepInput, state: Partial<AddClusterState>) {
+        if (!state.saslOption) {
+            return;
+        }
+        state.saslOption.password = await input.showInputBox({
+            title: INPUT_TITLE,
+            step: input.getStepNumber(),
+            totalSteps: state.totalSteps,
+            value: state.saslOption.password || '',
+            prompt: ' Password'
+        });

```

Comment:
Password appears in clear text

Reply:
Is it really a problem?

Reply:
it is when performing/recording demos ;-)

Reply:
I understand that but is it possible to see the typed password?

Reply:
fixed